core/src/main/scala/quasar/destination/gbq/GBQFlow.scala

```diff
@@ -246,11 +250,20 @@ final class GBQFlow[F[_]: Concurrent](
       Resource.eval(jobReq).flatMap(client.run).use { resp =>
         resp.status match {
           case Status.Ok => resp.headers match {
-            case Location(loc) => loc.uri.asRight[InitializationError[Json]].pure[F]
+            case Location(loc) => 
+              Sync[F].delay(log.debug(s"Successfully initialised job."))
+                .as(loc.uri.asRight[InitializationError[Json]])
           }
-          case _ =>  DestinationError.malformedConfiguration(
-            (GBQDestinationModule.destinationType, jString("Reason: " + resp.status.reason),
-            config.sanitizedJson.toString)).asLeft[Uri].pure[F]
+          case otherStatus =>  
+            resp.attemptAs[String].fold(
+                _ => Sync[F].delay(log.error(s"GBQ job creation failed with status '$otherStatus' and no body")),
+                body => Sync[F].delay(log.error(s"GBQ job creation failed with status '$otherStatus': $body")))
+              .as(DestinationError.invalidConfiguration(
+                (GBQDestinationModule.destinationType, 
+                  config.sanitizedJson,
+                  ZNEList("Reason: " + resp.status.reason))).asLeft[Uri])
```

Comment:
It is helpful if the reason is a short message that can be interpreted without further context, e.g. in here it could be `Error creating GBQ job`.

It won't be shown directly to the user, but it would be included in the file that users are asked to send to support in case a user does not know what is wrong (so we can then use this as a search term).

(similar for the other `invalidConfiguration`)